EmojiData/src/main/java/com/gluonhq/emoji/EmojiData.java

```diff
@@ -27,22 +27,42 @@ public class EmojiData {
      */
     private static final Map<String, Emoji> EMOJI_MAP = new HashMap<>();
 
+    /**
+     * Stores emojis with key's as their unified field
+     */
+    private static final Map<String, Emoji> EMOJI_UNICODE_MAP = new HashMap<>();
+
     static  {
+        // https://github.com/iamcal/emoji-data/blob/master/emoji.json
         try (final InputStream emojiStream = EmojiData.class.getResourceAsStream("emoji.json")) {
             InputDataSource dataSource = new BasicInputDataSource(emojiStream);
-            InputStreamIterableInputConverter<Emoji> converter = new JsonIterableInputConverter<>(Emoji.class);
+            InputStreamIterableInputConverter<Emoji> converter = new EmojiIterableInputConverter(Emoji.class);
             ListDataReader<Emoji> listDataReader = new InputStreamListDataReader<>(dataSource, converter);
             for (Iterator<Emoji> it = listDataReader.iterator(); it.hasNext();) {
                 Emoji e = it.next();
-                if (e.getShort_name().isPresent()) {
-                    EMOJI_MAP.put(e.getShort_name().get(), e);
+                EMOJI_UNICODE_MAP.put(e.getUnified(), e);
+                e.getShort_name().ifPresent(s -> EMOJI_MAP.put(s, e));
+                if (e.getSkin_variations() != null) {
+                    e.getSkin_variations().values()
+                            .forEach(v -> {
+                                EMOJI_UNICODE_MAP.put(v.getUnified(), v);
+                                v.getShort_name().ifPresent(s -> EMOJI_MAP.put(s, v));
+                            });
                 }
             }
         } catch (IOException e) {
             System.err.println("Could not load emoji json file" + e.getMessage());
-//            throw new IllegalStateException("Could not load emoji json file" + e.getMessage());
         }
     }
+
+    public static Optional<Emoji> emojiFromUnicode(String unicodeText) {
+        Emoji value = EMOJI_UNICODE_MAP.get(unicodeText);
+        if (value == null) {
+            // try to qualify it
+            value = EMOJI_UNICODE_MAP.get(unicodeText + "-FE0F");
```

Comment:
What exactly is going on here?

Reply:
As you can see in https://unicode.org/Public/emoji/14.0/emoji-test.txt, an emoji can be fully qualified, like ☺️ `\u263A\uFE0F` or unqualified like ☺ `\u263A`.

In this case, the emoji.json has:
```
{"name":"WHITE SMILING FACE","unified":"263A-FE0F","non_qualified":"263A", "docomo":"E6F0","au":"E4FB","softbank":"E414","google":"FE336","image":"263a-fe0f.png" ...
```
The Demo class has a long string with the unicodes for 3500+ emojis. In many cases, the unicode is the unqualified one, so instead of fixing that huge demo string, I added this small fix here, which doesn't hurt.